org-code-javabuilder/lib/src/main/java/org/code/javabuilder/CodeBuilderRunnable.java

```diff
@@ -86,16 +86,12 @@ private void executeCodeBuilder() {
 
       // Errors we didn't catch. These may have been caused by the JVM, our own setup, or many other
       // unknowns. Log them so we can fix them.
-      JSONObject eventData = new JSONObject();
-      eventData.put(LoggerConstants.EXCEPTION_MESSAGE, error.getExceptionMessage());
-      eventData.put(LoggerConstants.LOGGING_STRING, error.getLoggingString());
-      Logger.getLogger(MAIN_LOGGER).severe(eventData.toString());
+      LoggerUtils.logError(error);
 
       // Additionally, these may have affected the user. For now, let's tell them about it.
       outputAdapter.sendMessage(error.getExceptionMessage());
-    } finally {
-      GlobalProtocol.getInstance().cleanUpResources();
-      this.outputAdapter.sendMessage(new StatusMessage(StatusMessageKey.EXITED));
     }
+    // Send exit message if we haven't already returned
```

Comment:
nit: this comment is a little confusing. Maybe add why we don't need to send an exit message for cases when we return early?